GitVersionCore/GitPreparer.cs

```diff
@@ -202,10 +202,11 @@ private static Reference GetLocalReference(Repository repository, string branchN
             return repository.Refs.FirstOrDefault(localRef => string.Equals(localRef.CanonicalName, targetBranchName));
         }
 
-        private static DirectReference GetRemoteReference(Repository repository, string branchName, string repositoryUrl)
+        private static DirectReference GetRemoteReference(Repository repository, string branchName, Authentication authentication)
         {
             var targetBranchName = branchName.GetCanonicalBranchName();
-            var remoteReferences = repository.Network.ListReferences(repositoryUrl);
+            var remote = repository.Network.Remotes.Single();
```

Comment:
Are you sure the remote you're after will always exist? If that's not the case, I'd go a safer way, creating a temporary remote (with a random name) and the provided url, list remotes refs through it, then drop this temp remote.